addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/ContentSecurityPolicyScanRule.java

```diff
@@ -450,40 +404,166 @@ public int getWascId() {
         return 15; // WASC-15: Application Misconfiguration
     }
 
-    private void raiseAlert(
-            String name,
-            String otherInfo,
-            String param,
-            int risk,
-            String evidence,
-            String alertRef) {
-        raiseAlert(name, otherInfo, param, risk, evidence, getReference(), alertRef);
+    private AlertBuilder getBuilder() {
+        return newAlert()
+                .setConfidence(Alert.CONFIDENCE_HIGH)
+                .setDescription(Constant.messages.getString(MESSAGE_PREFIX + "desc"))
+                .setSolution(getSolution())
+                .setReference(getReference())
+                .setCweId(getCweId())
+                .setWascId(getWascId());
+    }
+
+    private AlertBuilder buildXcspAlert(int risk, String evidence) {
+        return getBuilder()
+                .setName(
+                        getName()
+                                + ": "
+                                + Constant.messages.getString(MESSAGE_PREFIX + "xcsp.name"))
+                .setOtherInfo(Constant.messages.getString(MESSAGE_PREFIX + "xcsp.otherinfo"))
+                .setEvidence(evidence)
+                .setParam(HTTP_HEADER_XCSP)
```

Comment:
Why no longer using `getHeaderField`?

Reply:
Hmmm I don’t fully remember, probably because it was one less thing to pass. I can restore/standardize that.